File: src/C#/BehavioralPatterns/Mediator/MultipleExamples/CommunicationHub/Mediators/DirectMediator.cs
using MultipleExamples.CommunicationHub.Participants;
using System;
using System.Collections.Generic;
using System.Text;

namespace MultipleExamples.CommunicationHub.Mediators
{
    public class DirectMediator : IMediator
    {
        public void Notify(Participant sender, object senderArgs)
        {
            if (!(senderArgs is List<object> argsList))
            {
                return;
            }

            if (!(argsList[0] is Participant receiver))
            {
                return;
            }

            receiver.Receive(sender, argsList[1]);
        }
    }
}

File: src/JavaScript/BehavioralPatterns/Mediator/CommunicationHub/mediators/directMediator.js

const Mediator = require("./mediator");
const Participant = require("../participants/participant");

class DirectMediator extends Mediator {
  notify(sender, senderArgs) {
    if (!Array.isArray(senderArgs)) {
      return;
    }

    const [receiver, args] = senderArgs;

    if (!(receiver instanceof Participant)) {
      return;
    }

    receiver.receive(sender, args);
  }
}

module.exports = DirectMediator;